go/kbfs/libkbfs/tlf_handle.go

```diff
@@ -412,12 +412,28 @@ func (h *TlfHandle) ToFavorite() Favorite {
 	}
 }
 
+// FavoriteData converts a TlfHandle into favoriteData, suitable for
+// Favorites calls.
+func (h *TlfHandle) FavoriteData() favoriteData {
+	return favoriteData{
+		Name:       string(h.GetCanonicalName()),
+		FolderType: h.Type().FolderType(),
+		// TODO: verify this conversion
+		ID:      keybase1.TLFID(h.tlfID.String()),
+		Private: h.Type() != tlf.Public,
+		// TODO: find this team ID if we care about it
+		TeamID:       keybase1.TeamID(0),
```

Comment:
For TLFs backed by an implicit or regular team, you should be able to get the TeamID easily with something like:

```go
fd := favoriteData{...}
if h.IsBackedByTeam() {
    fd.TeamID = h.FirstResolvedWriter().AsTeamOrBust()
}
```

If it's not backed by an implicit team, it must be an old folder that we've failed to convert, or an old public folder (haven't converted those yet).  In those cases, I doubt we need a team ID at all.  I'd be surprised if we do.